### PR TITLE
Add dynamic matrix generation for sdk-for-net

### PR DESCRIPTION
**Pull request by hallipr**
- Add script to generate props files containing sets of service projects to build
- Add job yaml for creating job matrix
- Use sparse-checkout steps from eng/common to reduce per-agent overhead
- Use override file instead of project filter when building service.proj in azure-sdk-for-net

### CODE REVIEW

**Comment by hallipr** [comment]
This PR is pending changes in azure-sdk-for-net:
https://github.com/Azure/azure-sdk-for-net/pull/35644

**Comment by hallipr** [comment]
Per comment on previous PR by @m-nash:
@archerzz @chunyu3 @pshao25 since you guys are more familiar with our pipeline can you please review this?

**Comment by hallipr** [comment]
> This PR is pending changes in azure-sdk-for-net: [Azure/azure-sdk-for-net#35644](https://github.com/Azure/azure-sdk-for-net/pull/35644)

The azure-sdk-for-net PR merged.  This PR is ready for merge pending review

**Review by benbp — APPROVED** [review]

**Review by chunyu3 — APPROVED** [review]

**Review by m-nash — APPROVED** [review]

File: eng/pipelines/sdk-update.yml [hunk 1]
```diff
@@ -13,7 +13,6 @@ resources:
       type: github
       name: Azure/azure-sdk-tools
       endpoint: azure
-      ref: refs/tags/azure-sdk-tools_20220404.3
```
Comment (weshaggard) [inline eng/pipelines/sdk-update.yml:16]: Why was this removed? Looks like it was added back with https://github.com/Azure/autorest.csharp/commit/768a83a695500265d208e757d5ea0a6443ebfac2 which is good but we do want to keep these refs instead of floating.
